File: java/src/trie/FindPrefixTest.java
package trie;

import org.junit.Assert;
import org.junit.Test;

import java.util.Arrays;

public class FindPrefixTest {
	@Test
	public void testRandomCases() {
		for (int test = 0; test < 10_000; test++) {
			var td = new randomTestData();

			var actual = FindPrefix.hasPrefix(td.strings);
			if (actual != td.hasPrefix) {
				Assert.fail(String.format("hasPrefix(%s):\nwant %s\n got %s",
					Arrays.toString(td.strings), td.hasPrefix, actual));
			}
		}
	}
}

File: python/trie/find_prefix_test.py
from find_prefix import has_prefix
from trie_test_data import RandomTestData


def test_random_cases():
	for _ in range(10_000):
		td = RandomTestData()

		actual = has_prefix(td.strings)
		assert actual == td.has_prefix, \
			f'has_prefix({td.strings}):\nwant {td.has_prefix}\n got {actual}'
